packages/backend-auth/API.md

```diff
@@ -45,7 +44,7 @@ export type AuthLoginWithFactoryProps = Omit<AuthProps['loginWith'], 'externalPr
 };
 
 // @public
-export const defineAuth: (props: AmplifyAuthProps) => ConstructFactory<AmplifyAuth & ResourceProvider<AuthResources>>;
+export const defineAuth: (props: AmplifyAuthProps) => ConstructFactory<AmplifyAuth>;

```

Comment:
AmplifyAuth already implements ResourceProvider<AuthResources>

Reply:
Is the expectation here that all these factories return `ConstructFactory<ResourceProvider>`? If so, does that need to be a shared type among all the `defineResource` factory functions?